### PR TITLE
make buddy box explicit and add request buddy notification

### PR DESCRIPTION
**Pull request by grosser**
screen shot 2017-04-18 at 2 08 03 pm

... and tons of cleanup ... removed stubs from tests ... removed duplicate logic around delivery ... made channel auto-cleanup `#`

### CODE REVIEW

File: plugins/slack_webhooks/app/models/slack_webhook_notification.rb [hunk 1]
```diff
@@ -71,4 +70,27 @@ def risks_field
...
+
+  def deploy_callback_content
+    subject = "[#{@project.name}] #{@deploy.summary}"
+    controller = ActionController::Base.new
+    view = ActionView::Base.new(File.expand_path("../../views/samson_slack_webhooks", __FILE__), {}, controller)
+    show_prs = @deploy.pending? || @deploy.running?
+    status_emoji =
+      case @deploy.status
+      when 'pending' then ':stopwatch:'
+      when 'running' then ':truck::dash:'
```
Comment (henders) [inline plugins/slack_webhooks/app/models/slack_webhook_notification.rb]: :)

**Comment by henders** [comment]
So selecting `Buddy Request` autosends the buddy request, otherwise `Buddy Box` lets them write their own custom one?

**Comment by henders** [comment]
If it's as I understood above, then 👍

**Comment by grosser** [comment]
correct

**Pull request merged**